includes/routes/event/edit.php

```diff
@@ -41,23 +42,30 @@ public function handle( int $event_id ): void {
 		$create_delete_button          = false;
 		$visibility_delete_button      = 'inline-flex';
 
-		$stats_calculator = new Stats_Calculator();
-		if ( ! $stats_calculator->event_has_stats( $event ) ) {
-			$current_user = wp_get_current_user();
-			if ( $current_user->ID === $event->post_author || current_user_can( 'manage_options' ) ) {
-				$create_delete_button = true;
-			}
-		}
-
 		try {
-			$event_start = self::convertToTimezone( get_post_meta( $event_id, '_event_start', true ), $event_timezone );
-			$event_end   = self::convertToTimezone( get_post_meta( $event_id, '_event_end', true ), $event_timezone );
+			$event_start   = self::convertToTimezone( get_post_meta( $event_id, '_event_start', true ), $event_timezone );
+			$event_end     = self::convertToTimezone( get_post_meta( $event_id, '_event_end', true ), $event_timezone );
+			$now           = self::convertToTimezone( 'now', $event_timezone );
+			$event_end_utc = new DateTime( get_post_meta( $event_id, '_event_end', true ), new DateTimeZone( 'UTC' ) );
+			$now_utc       = new DateTime( 'now', new DateTimeZone( 'UTC' ) );
 		} catch ( Exception $e ) {
 			// phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
 			error_log( $e );
 			$this->die_with_error( esc_html__( 'Something is wrong.', 'gp-translation-events' ) );
 		}
 
+		if ( $now_utc > $event_end_utc ) {
+			$this->die_with_error( esc_html__( 'You cannot edit a finished event.', 'gp-translation-events' ), 403 );
```

Comment:
I think "past" would be more idiomatic than "finished".

```suggestion
			$this->die_with_error( esc_html__( 'You cannot edit a past event.', 'gp-translation-events' ), 403 );
```